aspnet/fundamentals/dependency-injection.rst

```diff
@@ -1,10 +1,277 @@
+
+In the sample for this article, there is a simple controller that displays character names, called ``CharacterController``. Its ``Index`` method displays the current list of characters that have been stored in the application, and initializes the collection with a handful of characters if none exist. Note that although this application uses Entity Framework and the ``ApplicationDbContext`` class for its persistence, none of that is apparent in the controller. Instead, the specific data access mechanism has been abstracted behind an interface, ``ICharacterRepository``, which follows the `repository pattern <http://deviq.com/repository-pattern/>`_. An instance of ``ICharacterRepository`` is requested via the constructor and assigned to a private field, which is then used to access characters as necessary.
+
+.. literalinclude:: dependency-injection/sample/src/DependencyInjectionSample/Controllers/CharactersController.cs
+	:language: c#
+	:linenos:
+	:lines: 1-32
+	:emphasize-lines: 10,12,14,20,23-27
+
+The `ICharacterRepository` simply defines the two methods the controller needs to work with `Character` instances.
+
+.. literalinclude:: dependency-injection/sample/src/DependencyInjectionSample/Interfaces/ICharacterRepository.cs
+	:language: c#
+	:linenos:
+
+This interface is in turn implemented by a concrete type, ``CharacterRepository``, that is used at runtime.
+
+.. literalinclude:: dependency-injection/sample/src/DependencyInjectionSample/Models/CharacterRepository.cs
+	:language: c#
+	:linenos:
+
+Note that ``CharacterRepository`` requests an ``ApplicationDbContext`` in its constructor. It is not unusual for dependency injection to be used in a chained fashion like this, with each requested dependency in turn requesting its own dependencies. The container is responsible for resolving all of the dependencies in the tree and returning the fully resolved object graph.
+
+In this case, both ``ICharacterRepository`` and in turn ``ApplicationDbContext`` must be registered with the services container in ``ConfigureServices`` in ``Startup``. ``ApplicationDbContext`` is configured via the call to the extension method ``AddEntityFramework`` which includes an extension for adding a ``DbContext`` (``AddDbContext<T>``). Registration of the repository is done at the bottom end of ``ConfigureServices``:
+
+.. literalinclude:: dependency-injection/sample/src/DependencyInjectionSample/Startup.cs
+	:language: c#
+	:lines: 48-54, 82-86
+	:linenos:
+	:dedent: 8
+	:emphasize-lines: 6, 11
+
+Entity Framework contexts should be added to the services container using the ``Scoped`` lifetime. This is taken care of automatically if you use the helper methods as shown above. Repositories that will make use of Entity Framework should use the same lifetime.
```

Comment:
Added a note. I do want to mention EF specifically because it's going to be an extremely common element used with DI (and maybe they only read this article and don't read the whole EF docs - it's important they know this bit).